src/solidity-frontend/solidity_convert.cpp

```diff
@@ -541,6 +550,192 @@ bool solidity_convertert::get_struct_class_method(
   return false;
 }
 
+//Used to create a struct from tuple
+//Struct will be populated with variables
+//The variables within the struct will not be initialized
+bool solidity_convertert::get_struct_tuple(
+  const nlohmann::json &struct_def)
+{
+  //This function should only be used for TupleExpressions
+  assert(struct_def["nodeType"].get<std::string>() == "TupleExpression");
+  
+  // 1. populate name and id
+  std::string id, name;
+  struct_typet t = struct_typet();
+
+  // Conversion of tuple to struct so tuple has no default name
+  // But we still need to get unique label
+  // e.g. "sol:@C@BASE@tuple#14"
+  std::string label = std::to_string(struct_def["id"].get<int>());
+  name = "tuple#" + label;
+  id = "sol:@C@" + current_contractName + "@" + name;
+
+  t.tag(id);  
+
+  // 2. Check if the struct symbol is already added to the context, do nothing if it is
+  // already in the context.
+  if (context.find_symbol(id) != nullptr)
+  {
+    std::cout << "Found Struct Symbol " << std::endl;
+    return false;  
+  }
+
+  // 3. populate location
+  locationt location_begin;
+  get_location_from_decl(struct_def, location_begin);
+
+  // 4. populate debug module name
+  std::string debug_modulename =
+    get_modulename_from_path(location_begin.file().as_string());
+  current_fileName = debug_modulename;
+
+  // 5. set symbol attributes for the struct
+  symbolt symbol;
+  get_default_symbol(symbol, debug_modulename, t, name, id, location_begin);
+
+  symbol.is_type = true;
+  symbolt &added_symbol = *move_symbol_to_context(symbol);
+
+  //Components consist of our variables used in our tuple
+  nlohmann::json ast_nodes;
+  ast_nodes = struct_def["components"];
+
+  std::cout << "Populating identifiers" << std::endl;
+  // 6. Populate the Identifiers 
+
+  // std::string contract_name;
+  // if(get_current_contract_name(struct_def, contract_name))
+  // {
+  //   return true;
+  // }
+
+  for (nlohmann::json::iterator itr = ast_nodes.begin(); itr != ast_nodes.end();
+      ++itr)
```

Comment:
I feel like you should use `get_expr()` to parse each component. See `CallExprClass` => `structConstructorCall`:
```c++
// popluate components
    for (unsigned int i = 0; i < inits.operands().size() && i < args.size();
         i++)
    {
      exprt init;
      if (get_expr(args.at(i), members.at(i)["typeDescriptions"], init))
        return true;

      const struct_union_typet::componentt *c =
        &to_struct_type(t).components().at(i);
      typet elem_type = c->type();

      solidity_gen_typecast(ns, init, elem_type);
      inits.operands().at(i) = init;
    }
```

Reply:
I think this is more suited towards RHS, since its calling a previous initialized struct and populating the values. But this is definitely useful for my current implementation, which is LHS. Thank you for the help!